baqend/getTestOverviewReport.ts

```diff
@@ -0,0 +1,135 @@
+import { EntityManager } from 'baqend';
+import { BulkTest, TestOverview } from 'baqend/commonjs/lib/model';
+import { Request, Response } from 'express';
+import { iterateQuery, extractOrigin, processOrigin } from './_helpers';
+
+interface DomainStatus {
+  [status: string]: number;
+}
+
+interface SortedDomain {
+  domain: string;
+  count: number;
+  statuses: DomainStatus;
+}
+
+const EXCLUDED_DOMAINS = [
+  'kicker.de',
+  'plesk.com',
+  'speed-kit-test.com'
+];
+
+/**
+ * Fetches all test overview IDs within bulk tests for the specified date range.
+ *
+ * @param {EntityManager} db - The database entity manager.
+ * @param {Date} startDate - The start date of the range.
+ * @param {Date} endDate - The end date of the range.
+ * @returns {Promise<Set<string>>} - A promise that resolves to a set of test overview IDs.
+ */
+async function fetchAllTestOverviewIdsFromBulkTests(
+  db: EntityManager,
+  startDate: Date,
+  endDate: Date
+): Promise<Set<string>> {
+  const builder = db.BulkTest.find()
+    .greaterThanOrEqualTo('createdAt', startDate)
+    .lessThan('createdAt', endDate)
+
+  const result = await iterateQuery<BulkTest>(db, builder);
+  return new Set(result.flatMap((bulkTest: BulkTest) => bulkTest.testOverviews.map((overview: TestOverview) => overview.id).filter((id): id is string => id !== null)));
```

Comment:
Ich glaube den `id` filter können wir uns sparen. In der Liste sind die Ids immer da, sonst wären die nicht in der Liste.